src/labone/mock/automatic_hpk_functionality.py

```diff
@@ -0,0 +1,273 @@
+"""Partially predifined behaviour for HPK mock.
+
+This class provides basic Hpk mock functionality by taking over some usually
+desired tasks. With that in place, the user may inherit from this class
+in order to further specify behavior, without having to start from scratch.
+Even if some of the predefined behaviour is not desired, the implementation
+can give some reference on how an individual mock server can be implemented.
+
+
+Already predefined behaviour:
+
+    Simulating state for get/set:
+        A dictionary is used to store the state of the mock server.
+        Get and set will access this dictionary.
+
+    Answering list_nodes(_info) via knowledge of the tree structure:
+        Given a dictionary of paths to node info passed in the constructor,
+        the list_nodes(_info) methods will be able to answer accordingly.
+
+    Reducing get_with_expression/set_with_expression to multiple get/set:
+        As the tree structure is known, the get_with_expression/set_with_expression
+        methods can be implemented by calling the get/set methods multiple times.
+
+    Managing subscriptions and passing all changes into the queues:
+        The subscriptions are stored and on every change, the new value is passed
+        into the queues.
+
+    Adding chronological timestamps to responses:
+        The server answers need timestamps to the responsis in any case.
+        By using the monotonic clock, the timestamps are added automatically.
+
+"""
+
+from __future__ import annotations
+
+import asyncio
+import fnmatch
+import re
+import time
+import typing as t
+
+from labone.core import ListNodesFlags, ListNodesInfoFlags
+from labone.core.value import AnnotatedValue, Value
+from labone.mock.hpk_functionality import HpkMockFunctionality
+
+if t.TYPE_CHECKING:
+    from labone.core.helper import LabOneNodePath
+    from labone.core.session import NodeInfo
+    from labone.core.subscription import StreamingHandle
+
+
+class AutomaticHpkFunctionality(HpkMockFunctionality):
+    """Predefined behaviour for HPK mock.
+
+    See module docstring for details.
+
+    For customizations, inherit from this class, override the methods
+    and call super() in the overriden methods if desired.
```

Comment:
`super()` is bad practice ... I would just remove this section. It's not really adding much value.

Reply:
ok removed. But why is super() bad practice? How else would you extend the functionality?

Reply:
Bad practice might be a bit harsh ... its dangerous if you are not aware of the problems that come with it ... 
The general rule is

> Super usage has to be consistent: In a class hierarchy, super should be used everywhere or nowhere. Mixing super and classic calls is a confusing practice. People tend to avoid super, for their code to be more explicit. 

(quoted from Expert Python Programming)

The main problem is when you have multiple parent classes (which is also bad practice 🙈 ... but that's a different story)

IMHO super is misused so much that is easier to read if one does not use it .